exporter/containerimage/exptypes/types.go

```diff
@@ -23,3 +40,64 @@ type Platform struct {
 	ID       string
 	Platform ocispecs.Platform
 }
+
+func ExporterAnnotationIndexKey(key string) string {
+	return exporterAnnotationKey(AnnotationIndex, nil, key)
+}
+
+func ExporterAnnotationIndexDescriptorKey(key string) string {
+	return exporterAnnotationKey(AnnotationIndexDescriptor, nil, key)
+}
+
+func ExporterAnnotationManifestKey(p *ocispecs.Platform, key string) string {
+	return exporterAnnotationKey(AnnotationManifest, p, key)
+}
+
+func ExporterAnnotationManifestDescriptorKey(p *ocispecs.Platform, key string) string {
+	return exporterAnnotationKey(AnnotationManifestDescriptor, p, key)
+}
+
+func exporterAnnotationKey(annotationType string, platform *ocispecs.Platform, key string) string {
```

Comment:
Could define a type for `annotationType` for extra safety. Keep private if nothing outside uses it.

Reply:
We can do this with `type annotationType string` but it doesn't add much extra safety since go will allow implicitly converting between `string` and `annotationType` (because it's an alias)
If we create a `struct` to do this and actually construct a new type, then `switch`-statements no longer work.